browser/loaders_test.py

```diff
@@ -0,0 +1,175 @@
+"""Tests for loaders.py"""
+
+import io
+import json
+import tempfile
+import tarfile
+
+import pytest
+import numpy as np
+from PIL import Image
+
+import loaders
+
+
+def test_load_raw_npz():
+    """Creates a dummy NPZ file with just a raw array and loads it."""
+    expected = np.zeros((1, 1, 1, 1))
+    npz = io.BytesIO()
+    np.savez(npz, X=expected)
+    npz.seek(0)
+
+    loader = loaders.Loader()
+    loader._load_npz(npz)
```

Comment:
Could we test `_load` itself by replacing all of the `loader._load_X` calls to `loader._load`?

Reply:
Yes, if we also provide a filename with the right extension for the test data. Easy fix, and it makes the tests more consistent with how we'd use the loader, so I'll make this change.

Reply:
This is making me think a bit more closely about the Loader class attributes... Right now, `_load` raise a `NotImplementedError` because the a Loader doesn't have access to the data to load to. We expect each implementation of the Loader interface to retrieve the data somehow (from S3, or a file attachment, or the file system, etc.) and then they will implement `_load` and call the appropriate load function, passing the data as an argument.

I'm thinking the Loader instead should have a private `_data` attribute defined in the parent class, and the load functions will use this attribute instead of taking a `data` argument. Then, we can implement `_load` in the base class so it uses the `_data` attribute, which is something that normally would be populated by an implementation, but we can fill it with testing data instead within the tests.

Reply:
Reworked now, take a look and let me know what you think

Reply:
Looks awesome!